pulsar-broker-auth-sasl/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderSasl.java

```diff
@@ -175,6 +186,24 @@ private String createAuthRoleToken(String role, String sessionId) {
         return signed;
     }
 
+    private byte[] readSecretFromUrl(String secretConfUrl) throws IOException {
+        if (secretConfUrl.startsWith("file:")) {
+            try {
+                return IOUtils.toByteArray(URL.createURL(secretConfUrl.trim()));
```

Comment:
for JWT token Authentication we use URI and not URL
https://github.com/apache/pulsar/blob/330fcb9787d9f822667f0617b22bcae66b4644e5/pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationToken.java#L81

I don't know the exact difference but I believe it is better to do the same way